File: pathfinder-editmode/src/main/java/de/cubbossa/pathfinder/editmode/RoadMapEditorExtension.java
package de.cubbossa.pathfinder.editmode;

import de.cubbossa.cliententities.lib.packetevents.api.PacketEvents;
import de.cubbossa.cliententities.lib.packetevents.impl.factory.spigot.SpigotPacketEventsBuilder;
import de.cubbossa.menuframework.GUIHandler;
import de.cubbossa.pathfinder.AbstractPathFinder;
import de.cubbossa.pathfinder.BukkitPathFinder;
import de.cubbossa.pathfinder.PathFinder;
import de.cubbossa.pathfinder.PathFinderExtension;
import de.cubbossa.pathfinder.PathFinderExtensionBase;
import de.cubbossa.pathfinder.PathFinderPlugin;
import de.cubbossa.pathfinder.misc.NamespacedKey;
import org.jetbrains.annotations.NotNull;
import org.pf4j.Extension;

@Extension(points = PathFinderExtension.class)
public class RoadMapEditorExtension extends PathFinderExtensionBase implements PathFinderExtension {

  @NotNull
  @Override
  public NamespacedKey getKey() {
    return AbstractPathFinder.pathfinder("group-editor");
  }

  @Override
  public void onLoad(PathFinder pathPlugin) {
    if (pathPlugin instanceof BukkitPathFinder bukkitPathFinder) {
      PacketEvents.setAPI(SpigotPacketEventsBuilder.build(bukkitPathFinder.getJavaPlugin()));
      PacketEvents.getAPI().getSettings()
          .checkForUpdates(true)
          .bStats(true);
      PacketEvents.getAPI().load();
    }
  }

  @Override
  public void onEnable(PathFinder plugin) {

    PacketEvents.getAPI().init();

    new GUIHandler(PathFinderPlugin.getInstance());
    GUIHandler.getInstance().enable();
  }

  @Override
  public void onDisable(PathFinder plugin) {
    GUIHandler.getInstance().disable();

    PacketEvents.getAPI().terminate();
  }
}

File: pathfinder-editmode/src/main/kotlin/de/cubbossa/pathfinder/editmode/RoadMapEditorExtension.kt
package de.cubbossa.pathfinder.editmode

import de.cubbossa.cliententities.lib.packetevents.api.PacketEvents
import de.cubbossa.cliententities.lib.packetevents.impl.factory.spigot.SpigotPacketEventsBuilder
import de.cubbossa.menuframework.GUIHandler
import de.cubbossa.pathfinder.*
import de.cubbossa.pathfinder.misc.NamespacedKey
import org.pf4j.Extension

@Extension(points = [PathFinderExtension::class])
class RoadMapEditorExtension : PathFinderExtensionBase(), PathFinderExtension {
    override fun getKey(): NamespacedKey {
        return AbstractPathFinder.pathfinder("group-editor")
    }

    override fun onLoad(pathPlugin: PathFinder) {
        if (pathPlugin is BukkitPathFinder) {
            PacketEvents.setAPI(
                SpigotPacketEventsBuilder
                    .build(pathPlugin.javaPlugin)
            )
            com.github.retrooper.packetevents.PacketEvents.setAPI(
                io.github.retrooper.packetevents.factory.spigot.SpigotPacketEventsBuilder.build(
                    pathPlugin.javaPlugin
                )
            )
            com.github.retrooper.packetevents.PacketEvents.getAPI().settings
                .checkForUpdates(true)
                .bStats(true)
            com.github.retrooper.packetevents.PacketEvents.getAPI().load()
        }
    }

    override fun onEnable(plugin: PathFinder) {
        com.github.retrooper.packetevents.PacketEvents.getAPI().init()

        GUIHandler(PathFinderPlugin.getInstance())
        GUIHandler.getInstance().enable()
    }

    override fun onDisable(plugin: PathFinder) {
        GUIHandler.getInstance().disable()

        com.github.retrooper.packetevents.PacketEvents.getAPI().terminate()
    }
}
